src/Document.php

```
@@ -131,14 +136,20 @@ private static function buildQuery(string $startQuery, array $attributes): strin
      */
     public function select(string $query, ?array $attributes = null, ?DOMNode $context = null): QueryResult
     {
-        if (!empty($attributes)) {
+        if ($attributes !== null && $attributes !== []) {
             $query = self::buildQuery($query, $attributes);
         }
 
-        return new QueryResult($this->xpath->query($query, $context), $this->extractor);
+        $result = $this->xpath->query($query, $context);
+        if ($result === false) {
+            $result = new \DOMNodeList();
```

PHPMD (2.15.0)

Missing class import via use statement (line '150', column '27'). (undefined)

(MissingImport)